### ISSUE TITLE
Bundle validation fails if CSV contains multiple entries for one owned CRD at different api versions

### ISSUE THREAD

**Issue by komish** [operator-framework/operator-sdk#3146]
## Bug Report

**What did you do?**

1) Create a New Project

```
$ operator-sdk new someservice-operator --repo github.com/example/someservice-operator
INFO[0000] Creating new Go operator 'someservice-operator'. 
INFO[0000] Created go.mod                               
INFO[0000] Created tools.go                             
INFO[0000] Created cmd/manager/main.go                  
INFO[0000] Created build/Dockerfile                     
INFO[0000] Created build/bin/entrypoint                 
INFO[0000] Created build/bin/user_setup                 
INFO[0000] Created deploy/service_account.yaml          
INFO[0000] Created deploy/role.yaml                     
INFO[0000] Created deploy/role_binding.yaml             
INFO[0000] Created deploy/operator.yaml                 
INFO[0000] Created pkg/apis/apis.go                     
INFO[0000] Created pkg/controller/controller.go         
INFO[0000] Created version/version.go                   
INFO[0000] Created .gitignore                           
INFO[0000] Validating project                           
INFO[0004] Project validation successful.               
INFO[0004] Project creation complete.  

$ cd someservice-operator/
```
2) Create the initial example.com/v1beta1 API
```
$ operator-sdk add api --kind SomeService --api-version example.com/v1beta1
INFO[0000] Generating api version example.com/v1beta1 for kind SomeService. 
INFO[0000] Created pkg/apis/example/group.go            
INFO[0002] Created pkg/apis/example/v1beta1/someservice_types.go 
INFO[0002] Created pkg/apis/addtoscheme_example_v1beta1.go 
INFO[0002] Created pkg/apis/example/v1beta1/register.go 
INFO[0002] Created pkg/apis/example/v1beta1/doc.go      
INFO[0002] Created deploy/crds/example.com_v1beta1_someservice_cr.yaml 
INFO[0002] Running deepcopy code-generation for Custom Resource group versions: [example:[v1beta1], ] 
INFO[0011] Code-generation complete.                    
INFO[0011] Running CRD generator.                       
INFO[0011] CRD generation complete.                     
INFO[0011] API generation complete.
```
3) Mark this API as the stored version in `pkg/apis/example/v1beta1/someservice_types.go` as described here https://book.kubebuilder.io/multiversion-tutorial/api-changes.html#storage-versions

added `+kubebuilder:storageversion`
```
// SomeService is the Schema for the someservices API
// +kubebuilder:subresource:status
// +kubebuilder:resource:path=someservices,scope=Namespaced
// +kubebuilder:storageversion
type SomeService struct {
    metav1.TypeMeta   `json:",inline"`
    metav1.ObjectMeta `json:"metadata,omitempty"`

    Spec   SomeServiceSpec   `json:"spec,omitempty"`
    Status SomeServiceStatus `json:"status,omitempty"`
}
```

4) Add another API version for this resource
```
$operator-sdk add api --kind SomeService --api-version example.com/v1beta2
INFO[0000] Generating api version example.com/v1beta2 for kind SomeService. 
INFO[0002] Created pkg/apis/example/v1beta2/someservice_types.go 
INFO[0002] Created pkg/apis/addtoscheme_example_v1beta2.go 
INFO[0002] Created pkg/apis/example/v1beta2/register.go 
INFO[0002] Created pkg/apis/example/v1beta2/doc.go      
INFO[0002] Created deploy/crds/example.com_v1beta2_someservice_cr.yaml 
INFO[0002] Running deepcopy code-generation for Custom Resource group versions: [example:[v1beta1 v1beta2], ] 
INFO[0019] Code-generation complete.                    
INFO[0019] Running CRD generator.                       
INFO[0020] CRD generation complete.                     
INFO[0020] API generation complete.        
```

5) Generate a CSV

I end up with empty version strings here so I just edited the version from `someservice-operator.v` to `somesevice-operator.v0.1.0` so that the bundle validation doesn't complain about it since it's irrelevant to this bug.

```
$ operator-sdk generate csv
INFO[0000] Generating CSV manifest version              
INFO[0017] Fill in the following required fields in file someservice-operator.clusterserviceversion.yaml:
	spec.description
	spec.keywords
	spec.provider 
INFO[0017] CSV manifest generated successfully 
```
6) Build the bundle
```
$ operator-sdk bundle create -d deploy/olm-catalog/someservice-operator/manifests/ --generate-only
INFO[0000] Building annotations.yaml                    
INFO[0000] Building Dockerfile      
```
7) Validate the bundle and observe the error
```
$ operator-sdk bundle validate deploy/olm-catalog/someservice-operator/
FATA[0000] Bundle content validation failed: Bundle validation errors: Error: Value someservices.example.com: owned CRD "someservices.example.com" not found in bundle "someservice-operator.v0.1.0"  bundle-dir=/Users/myusername/scratch/someservice-operator/deploy/olm-catalog/someservice-operator
```

The owned CRDs are listed as follows in the CSV
```
spec:
  apiservicedefinitions: {}
  customresourcedefinitions:
    owned:
    - description: SomeService is the Schema for the someservices API
      kind: SomeService
      name: someservices.example.com
      version: v1beta2
    - description: SomeService is the Schema for the someservices API
      kind: SomeService
      name: someservices.example.com
      version: v1beta1
```
**What did you expect to see?**

I expected the bundle to validate even if with a single a CRD exposing a kind at multiple versions. 

Not sure if that means fixing how things are represented in the CSV (should we have multiple entries for one Group/Kind at multiple api-versions?) or if this is just an expectation that's incorrect in the bundle validation logic. Couldn't find documentation on it, so it could be documented and I just missed it.

**What did you see instead? Under which circumstances?**

The bundle validation complains (somewhat unclearly) that it cannot find a matching CRD yaml file in the bundle associated with an owned CRD.

**Environment**
* operator-sdk version:
```operator-sdk version: "v0.17.1", commit: "bb0c8f0108d1fd1be896d66c2d9e9044f10565a0", kubernetes version: "unknown", go version: "go1.13.10 darwin/amd64"```

* go version:

```
go version go1.13 darwin/amd64
```
* Kubernetes version information:
N/A

* Kubernetes cluster kind: 
N/A
* Are you writing your operator in ansible, helm, or go?
Go
**Possible Solution**

N/A
**Additional context**

**Comment by estroz** [comment]
@komish I think this is fixed in v0.18.0 in which we bumped our validation library version.

/triage support

**Comment by komish** [comment]
@estroz Thanks! Just saw this come up in bug triage. I'll test and report back if anything else, or close if not.

**Comment by komish** [comment]
```
[someservice-operator]$ operator-sdk version
operator-sdk version: "v0.18.0", commit: "921c33982687d6e9219fe0a1a326515fb37ae040", kubernetes version: "v1.18.2", go version: "go1.13.10 darwin/amd64"
```

@estroz Seems like `bundle validate` isn't seeing either CRD now?

```
[someservice-operator]$ operator-sdk bundle validate deploy/olm-catalog/someservice-operator/ --verbose
DEBU[0000] Debug logging is set                         
DEBU[0000] Found manifests directory                     bundle-dir=deploy/olm-catalog/someservice-operator container-tool=docker
DEBU[0000] Found metadata directory                      bundle-dir=deploy/olm-catalog/someservice-operator container-tool=docker
DEBU[0000] Getting mediaType info from manifests directory  bundle-dir=deploy/olm-catalog/someservice-operator container-tool=docker
INFO[0000] Found annotations file                        bundle-dir=deploy/olm-catalog/someservice-operator container-tool=docker
INFO[0000] Could not find optional dependencies file     bundle-dir=deploy/olm-catalog/someservice-operator container-tool=docker
DEBU[0000] Validating bundle contents                    bundle-dir=deploy/olm-catalog/someservice-operator container-tool=docker
ERRO[0000] Error: Value someservices.example.com: owned CRD "someservices.example.com" not found in bundle "someservice-operator.v0.0.1" 
ERRO[0000] Error: Value someservices.example.com: owned CRD "someservices.example.com" not found in bundle "someservice-operator.v0.0.1" 
```
The files
```
[someservice-operator]$ tree deploy/olm-catalog/someservice-operator/
deploy/olm-catalog/someservice-operator/
├── manifests
│   ├── example.com_someservices_crd.yaml
│   └── someservice-operator.clusterserviceversion.yaml
└── metadata
    └── annotations.yaml

2 directories, 3 files
```

I also tried this with only a single API version and it's not seeing the CRD. Something going on with the generation logic, or user error on my part that I'm missing?

```
[someservice-operator]$ operator-sdk bundle validate deploy/olm-catalog/someservice-operator/
INFO[0000] Found annotations file                        bundle-dir=deploy/olm-catalog/someservice-operator container-tool=docker
INFO[0000] Could not find optional dependencies file     bundle-dir=deploy/olm-catalog/someservice-operator container-tool=docker
ERRO[0000] Error: Value someservices.example.com: owned CRD "someservices.example.com" not found in bundle "someservice-operator.v0.0.1"
```

**Comment by estroz** [comment]
Can confirm this is a bug

/kind bug
/remove-triage support

**Comment by estroz** [comment]
There are at least 2 bugs in the validation library we use:
1. Only `v1beta1` CRDs are validated and the SDK generates `v1` CRDs by default, which is the source of this bug.
1. https://github.com/operator-framework/api/pull/32 was reverted, such that multiple CRD versions per CSV cause "duplicate CRD" errors.

I'll fix this upstream and release v0.18.1.
